test/unit/test_unit.ml

```diff
@@ -0,0 +1,100 @@
+open Base
+open Ocamlformat_lib
+
+let location_tests =
+  let open Migrate_ast in
+  let test_compare_width_decreasing name a b expected =
+    ( "compare_width_decreasing: " ^ name
+    , `Quick
+    , fun () ->
+        let got = Location.compare_width_decreasing a b in
+        Alcotest.check Alcotest.int Caml.__LOC__ expected got )
+  in
+  let make_pos pos_cnum =
+    {Lexing.pos_fname= "fname"; pos_lnum= 0; pos_bol= 0; pos_cnum}
+  in
+  let make_loc s e loc_ghost =
+    {Location.loc_start= make_pos s; loc_end= make_pos e; loc_ghost}
+  in
+  [ test_compare_width_decreasing "equal" (make_loc 0 0 false)
+      (make_loc 0 0 false) 0
+  ; test_compare_width_decreasing "different start" (make_loc 0 0 false)
+      (make_loc 1 0 false) (-1)
+  ; test_compare_width_decreasing "same start" (make_loc 0 0 false)
+      (make_loc 0 1 false) 1
+  ; test_compare_width_decreasing "same start, same end" (make_loc 0 0 true)
+      (make_loc 0 0 false) 1 ]
+
+let noit_tests =
+  let module Itv = struct
+    module T = struct
+      type t = int * int
+
+      let sexp_of_t (s, e) =
+        Base.Sexp.List [Base.Int.sexp_of_t s; Base.Int.sexp_of_t e]
+
+      let compare = Poly.compare
+    end
+
+    include T
+    include Comparator.Make (T)
+
+    let contains (sa, ea) (sb, eb) = sa <= sb && eb <= ea
+
+    let compare_width_decreasing (sa, ea) (sb, eb) =
+      Poly.compare (sa, -ea) (sb, -eb)
+
+    let pp ppf (s, e) = Caml.Format.fprintf ppf "(%d, %d)" s e
+
+    let equal = Poly.equal
+  end in
+  let module T = Non_overlapping_interval_tree.Make (Itv) in
+  let to_string t = Format_.asprintf "%a" Fmt.eval (T.dump t) in
+  let itv_list = Alcotest.list (module Itv) in
+  let test_dump name l expected =
+    ( "dump: " ^ name
+    , `Quick
+    , fun () ->
+        let t = T.of_list l in
+        let got = to_string t in
+        Alcotest.check Alcotest.string Caml.__LOC__ expected got )
+  in
+  let test_roots name l expected =
+    ( "roots: " ^ name
+    , `Quick
+    , fun () ->
+        let t = T.of_list l in
+        let got = T.roots t in
+        Alcotest.check itv_list Caml.__LOC__ expected got )
+  in
+  let test_children name l i expected =
+    ( "children: " ^ name
+    , `Quick
+    , fun () ->
+        let t = T.of_list l in
+        let got = T.children t i in
+        Alcotest.check itv_list Caml.__LOC__ expected got )
+  in
+  [ test_dump "empty" [] ""
```

Comment:
@NathanReb how do you structure alcotest suites these days? does that look ok or do you think it's better to pull functions and use `List.concat [test_dump; test_roots; test_children]`?

Reply:
Usually the way I do it is that for each function under test I maintain a list of test-cases `(name * speed * unit -> unit)`.

I then group them by modules or submodules by concatenating them. You can take a look at an example here: https://github.com/ocamllabs/dune-release/blob/master/tests/test_pkg.ml.

It requires a tiny bit of boiler plate but results in nice and helpful output from alcotest and works well with the `-e` option.

From a quick look that's pretty much what you're doing here with one exception, I pass the test case name to `Alcotest.check` instead of the loc as the loc will be the same for every test case and therefore won't be of much help when looking at the error report from Alcotest.

Reply:
Thanks, I just did this. Seems better indeed!